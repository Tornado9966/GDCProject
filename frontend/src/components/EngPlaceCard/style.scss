.ui.card {
    .content.place-card {
        .header.place-card-location {
            text-align: center;
            margin: 10px 0 15px;
            font-size: 1.4vw;
        }
        .image.place-card-image{
            width: 25.47vw;
            max-height: 13vw;
            overflow: hidden;
        }
        .meta.place-card-workingHours{
            text-align: center;
            font-size: 1vw;
            margin-top: 15px;
        }
    }
    .ui.buttons {
        align-items: center;
    }
    .placeCard__order {
        border: 1px solid #fbbd08;
        background: #fff;
        border-radius: 3px;
    }
}

@media all and (max-width: 992px) {
    .ui.card {
        .content.place-card {
            .header.place-card-location {
                font-size: 4.4vw;
            }
            .image.place-card-image{
                width: 100%;
                max-height: 100%;
                overflow: hidden;
            }
            .meta.place-card-workingHours{
                font-size: 4vw;
            }
        }
    }
}