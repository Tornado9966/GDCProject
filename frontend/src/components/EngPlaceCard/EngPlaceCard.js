import React from 'react';
import {Button, Card, Image} from 'semantic-ui-react';
import {Link} from 'react-router-dom';
import {PropTypes} from 'prop-types';
import useOrder from 'context/useOrder';
import {viewOnMap, bookHere, workingHoursText} from 'constants/text-constants';
import defaultImage from './default.jpg';

import './style.scss';

export const EngPlaceCard = props => {
  const {_id, name, location, workingHours, imageSrc} = props;
  const {setRestaurantStep : chooseRestaurant} = useOrder();
  return (
    <Card key={_id}>
      <Card.Content className="place-card">
        <Card.Header className="place-card-location">{location}</Card.Header>
        <Image
          className="place-card-image"
          size='big'
          bordered
          src={imageSrc}
        />
        <Card.Meta className="place-card-workingHours">
          {workingHoursText+workingHours}
        </Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <div className='ui two buttons'>
          <Button basic as={Link} to='/restaurants'>
            {viewOnMap}
          </Button>
          <Button className="placeCard__order" basic as={Link} to='/order' onClick={() => chooseRestaurant(_id, name, imageSrc, location)}>
            {`${bookHere} ${name}`}
          </Button>
        </div>
      </Card.Content>
    </Card>
  );
};

EngPlaceCard.propTypes = {
  _id: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  workingHours: PropTypes.string.isRequired,
  imageSrc: PropTypes.string
};

EngPlaceCard.defaultProps = {
  imageSrc: defaultImage
};
