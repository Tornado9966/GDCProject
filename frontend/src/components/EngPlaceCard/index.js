import {EngPlaceCard} from './EngPlaceCard';

export default EngPlaceCard;
