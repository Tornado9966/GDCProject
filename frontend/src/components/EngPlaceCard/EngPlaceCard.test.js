import React from 'react';
import { Link, MemoryRouter } from 'react-router-dom';

import { shallow, mount } from 'enzyme';
import { EngPlaceCard } from './EngPlaceCard';

const setUp = (place) => {
  return shallow(<EngPlaceCard {...place} />);
};

describe('Registration component', () => {
  let component;
  const place = {
    _id : '1',
    name: 'Restaurant',
    location: 'Some Street',
    workingHours: '9am to 9 pm',
    imageSrc: 'https://firebasestorage.googleapis.com/v0/b/eatngo-fbbeb.appspot.com/o/photo_2019-07-10_09-55-43.jpg?alt=media&token=5c94c1c7-9c2c-41f1-ab7a-7f2173966c2d'
  };

  beforeEach(() => {
    component = setUp(place);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('it should contain place-card element', () => {
    const wrapper = component.find('.place-card');
    expect(wrapper.length).toBe(1);
  });

  it('it should contain place-card-location element with corresponding text', () => {
    const wrapper = component.find('.place-card-location');
    const location = [...wrapper][0].props.children;
    expect(wrapper.length).toBe(1);
    expect(location).toBe(place.location);
  });

  it('it should contain place-card-image element with corresponding image', () => {
    const wrapper = component.find('.place-card-image');
    const image = [...wrapper][0].props.src;
    expect(wrapper.length).toBe(1);
    expect(image).toBe(place.imageSrc);
  });

  it('it should contain place-card-workingHours element with corresponding text', () => {
    const wrapper = component.find('.place-card-workingHours');
    const workingHours = [...wrapper][0].props.children;
    expect(wrapper.length).toBe(1);
    expect(workingHours).toBe(`Working hours: ${place.workingHours}`);
  });

  it('it should contain div with two buttons', () => {
    const wrapper = component.find('.ui.two.buttons');
    expect(wrapper.length).toBe(1);
  });

  it('it should contain two buttons', () => {
    const wrapper = mount(
      <MemoryRouter>
        <EngPlaceCard {...place} />
      </MemoryRouter>);
    expect(wrapper.find('a[role="button"]').length).toBe(2);
  });

  it('it should contain two buttons with corresponding links', () => {
    const wrapper = mount(
      <MemoryRouter>
        <EngPlaceCard {...place} />
      </MemoryRouter>);

    const links = [...wrapper.find(Link)];
    expect(links[0].props.to).toBe('/restaurants');
    expect(links[1].props.to).toBe('/order');
  });
});
