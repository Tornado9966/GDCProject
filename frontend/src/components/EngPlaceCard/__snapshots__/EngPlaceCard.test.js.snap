// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EngPlaceCard should render correctly 1`] = `ShallowWrapper {}`;

exports[`EngPlaceCard should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`EngPlaceCard should render correctly in "debug" mode 2`] = `ShallowWrapper {}`;

exports[`EngPlaceCard should render correctly in "debug" mode 3`] = `ShallowWrapper {}`;

exports[`Registration component renders correctly 1`] = `ShallowWrapper {}`;
